### ISSUE TITLE
Hybrid search: Missing documents and inaccurate count value with a where clause

### ISSUE THREAD

**Issue by nicolastoulemont** [askorama/orama#730]
### Describe the bug

Using the hybrid search with a where clause and a offset + a limit has two issues:
- The count value is upperbounded to the limit
- Some documents are incorrectly not returned.

### To Reproduce

You can find a codesandbox with the issue [here](https://codesandbox.io/p/sandbox/objective-sun-kzs96j)

More details on the issue on slack [here](https://orama-community.slack.com/archives/C0445CLH631/p1717166829824649)

TLDR: The issue is due to pagination being apply twice with the hybrid search call-stack both [on the uniqueTokenScores](https://github.com/askorama/orama/blob/main/packages/orama/src/methods/search-hybrid.ts#L88) and within the fetchDocuments function.

### Expected behavior

Correct document fetching and correct count values.

### Environment Info

```bash
OS: MacOS
Node: v20.X
```

### Affected areas

Search
